app/src/main/java/chat/rocket/android/chatrooms/presentation/ChatRoomsPresenter.kt

```diff
@@ -5,26 +5,52 @@ import chat.rocket.android.server.domain.GetCurrentServerInteractor
 import chat.rocket.android.server.infraestructure.RocketChatClientFactory
 import chat.rocket.android.util.launchUI
 import chat.rocket.core.RocketChatClient
+import chat.rocket.core.internal.model.Subscription
+import chat.rocket.core.internal.realtime.*
 import chat.rocket.core.internal.rest.chatRooms
 import chat.rocket.core.model.ChatRoom
+import chat.rocket.core.model.Room
+import kotlinx.coroutines.experimental.CommonPool
+import kotlinx.coroutines.experimental.Job
+import kotlinx.coroutines.experimental.delay
+import kotlinx.coroutines.experimental.launch
+import timber.log.Timber
 import javax.inject.Inject
 
 class ChatRoomsPresenter @Inject constructor(private val view: ChatRoomsView,
                                              private val strategy: CancelStrategy,
                                              private val serverInteractor: GetCurrentServerInteractor,
                                              private val factory: RocketChatClientFactory) {
-    lateinit var client: RocketChatClient
+    private val client: RocketChatClient = factory.create(serverInteractor.get()!!)
 
-    fun chatRooms() {
-        // TODO - check for current server
-        client = factory.create(serverInteractor.get()!!)
+    // FIXME - cache rooms on memory while we don't have a Database...
+    private var chatRooms = ArrayList<ChatRoom>()
```

Comment:
I'm doing this on my PR [spotlight](https://github.com/RocketChat/Rocket.Chat.Android/pull/695)